src/ConfigManager.py

```diff
@@ -109,15 +123,37 @@ def query_module(self, module: Core, module_friendly_name: str) -> ConfigModule:
         :return: Config module with the choices of the user
         """
         module_config = module.get_parameter_options()
-        module_config = ConfigManager._query_for_config(module_config,
-                                                        f"(Option for {module.__class__.__name__})")
-        available_submodules = module.get_available_submodule_options()
-        submodule_answer = checkbox(key='submodules', message="What submodule do you want?",
-                                    choices=available_submodules)
+        full_module_config = ConfigManager._query_for_config(module_config, f"(Option for {module.__class__.__name__})")
+        depending_submodules = False
+        for parameter in module_config:
+            if ("depending_submodule" in module_config[parameter] and
+                    module_config[parameter]["depending_submodule"]):
+                available_submodules = module.get_available_submodules(full_module_config[parameter])
+                depending_submodules = True
+        if not depending_submodules:
+            available_submodules = module.get_available_submodule_options()
+        more_module_config = {}
```

Comment:
I think we need a comment explaining `module_config`, `more_module_config` and `full_module_config`.

Reply:
Yeah, good idea.